packages/jira-adapter/src/fix_elements/remove_missing_extension_transition_rules.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ *                      Copyright 2024 Salto Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable no-console */
+
+import _ from 'lodash'
+import { isInstanceElement, Element, ElemID } from '@salto-io/adapter-api'
+import { collections, values } from '@salto-io/lowerdash'
+import { FixElementsHandler } from './types'
+import { getInstalledExtensionsMap, ExtensionType } from '../common/extensions'
+import {
+  WorkflowV2TransitionConditionGroup,
+  WorkflowV2Transition,
+  WorkflowV2TransitionRule,
+  isWorkflowV2Instance,
+} from '../filters/workflowV2/types'
+import { HasKeyOfType, MakeOptional } from '../../../lowerdash/src/types'
+import { getExtensionIdFromWorkflowTransitionRule } from '../common/workflow/transition_rules'
+
+const { awu } = collections.asynciterable
+
+const updateAndCheckTransitionRules = <
+  K extends string,
+  T extends MakeOptional<HasKeyOfType<K, WorkflowV2TransitionRule[]>, K>,
+>(
+  obj: T,
+  key: K,
+  predicate: (value: WorkflowV2TransitionRule) => boolean,
+): boolean => {
```

Comment:
does this work? it's clearer IMO

```suggestion
  K extends string,
>(
  obj: Partial<HasKeyOfType<K, WorkflowV2TransitionRule[]>>,
  key: K,
  predicate: (value: WorkflowV2TransitionRule) => boolean,
): boolean => {
```

Reply:
Doesn't partial make all properties optional? I wanted to make only one optional.

Reply:
`HasKeyOfType` or this function doesn't know about the other fields of the input object, so it doesn't matter

Reply:
I guess you're right in this context, removed the type.